modules/fb_instant_articles_display/fb_instant_articles_display.module

```diff
@@ -5,6 +5,134 @@
  * Hook implementations for Facebook Instant Articles Display module.
  */
 
+/**
+ * Implements hook_entity_load().
+ */
+function fb_instant_articles_display_entity_load($entities, $type) {
+  $enabled_entity_types = fb_instant_articles_display_get_article_entity_types();
+  foreach ($entities as $entity) {
+    // Only proceed if the entity is of the configured entity type and bundle.
+    list($id, $vid, $bundle) = entity_extract_ids($type, $entity);
+    if (!in_array($type, array_keys($enabled_entity_types))
+      || !in_array($bundle, array_keys($enabled_entity_types[$type]))) {
+      return FALSE;
+    }
+
+    // We add Entity-specific information to the InstantArticle object, and
+    // attach it to the Entity for use in later hooks.
+    // @see fb_instant_articles_display_module_implements_alter()
+    $wrapper = \Drupal\fb_instant_articles\ArticleWrapper::create();
+    $wrapper->entity_type = $type;
+    $wrapper->entity = $entity;
+
+    $entity->fb_instant_article_wrapper = $wrapper;
+  }
+}
+
+/**
+ * Implements hook_field_formatter_view().
+ */
+function fb_instant_articles_display_field_formatter_view($entity_type, $entity, $field, $instance, $langcode, $items, $display) {
+  // Grab the InstantArticle object we're going to be building up. If it's not
+  // present, it means that a site admin selected an FB IA field formatter for
+  // a non-FB IA content type.
+  if (!isset($entity->fb_instant_article_wrapper)) {
+    return FALSE;
+  }
+
+  $article = $entity->fb_instant_article_wrapper->getArticle();
+  $entity_info = entity_get_info($entity_type);
+  $bundle_key = !empty($entity_info['entity keys']['bundle']) ? $entity_info['entity keys']['bundle'] : $entity_type;
```

Comment:
@m4olivei Yep good call. I had already fixed this in another test I'm doing. Just added it to this PR in a897974.